Query MediaStore for media file URI for setting ringtone

TranceLove opened the pull request:
## Description

With reference to #4367, by inspecting the ringtone settings of a device

```sh
settings get system notification_sound
```

It becomes known that ringtone requires a content URI to the selected audio file on the device.

This PR tries to create that URI by querying the MediaStore. Therefore there is a prerequisite that even the audio file is visible in Amaze, it must also be visible to the MediaStore too, that refreshing the MediaStore is required.

Had attempted to do this for end user in code, but only worked on Android 14 or above, so gave up and not reinvent the wheel instead.

#### Issue tracker   
Fixes #4367

#### Automatic tests
<!-- remember to do manual testing when making UI changes! -->
- [x] Added test cases
  
#### Manual tests
- [x] Done  

Device:
- Pixel 4 emulator running Android 9
- Pixel 7 emulator running Android 14

1. Make sure the audio file exists in MediaStore. Some apps like Musicolet can help refresh the MediaStore
2. Open settings -> Sound
3. Change ringtone
4. Choose Amaze when app selection pops up
5. Choose the audio file
6. Ringtone name should shown equal to filename without extension, and no Toast pops up
7. Call the device
8. Device should ring without crashing any app

#### Build tasks success  
Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

TranceLove committed:
Query MediaStore for media file URI for setting ringtone

Fixes #4367

TranceLove committed:
Query MediaStore for media file URI for setting ringtone

Fixes #4367

TranceLove committed:
Add unit tests for changes

TranceLove committed:
Add unit tests for changes

EmmanuelMess commented on app/src/main/java/com/amaze/filemanager/filesystem/MediaStoreHack.java:
```diff
@@ -164,6 +165,43 @@ public static OutputStream getOutputStream(Context context, String str) {
+      } else {
+        return null;
+      }
+    } catch (Exception e) {
```
You shouldn't catch a generic `Exception`, just the specific exception expected on this. Here it is `UnsupportedOperationException`.

EmmanuelMess commented on app/src/main/res/values/strings.xml:
```diff
@@ -875,5 +875,6 @@ You only need to do this once, until the next time you select a new location for
     <string name="trash_bin_cleanup_interval_title">Cleanup interval</string>
     <string name="trash_bin_cleanup_interval_summary">Trigger auto-cleanup interval (hours)</string>
     <string name="file_deletion">File Deletion</string>
+    <string name="error_mediastore_query_uri">Error querying MediaStore for content URI of selected media file. Refresh the MediaStore and try again.</string>
```
This text seems too long for a Toast, it gets cut on "Refresh the..." on Pixel 9 emulator.

EmmanuelMess reviewed (dismissed):
Tested on Pixel 9 API 34, sets the ringtone correctly.

TranceLove committed:
Changes with changes from upstream

Use MediaConnectionUtils.scanFiles(context, arrayOf(it))

TranceLove committed:
Shorten query MediaStore URI error message

Also made MediaStoreHack.getUriForMusicMediaInternal() to catch UnsupportedOperationException when content URI cannot be created

EmmanuelMess reviewed (approved):
Reapproving based on previous tests.
